packages/flutter/lib/src/widgets/dismissable.dart

```diff
@@ -64,7 +64,6 @@ class Dismissable extends StatefulWidget {
     this.onDismissed,
     this.direction: DismissDirection.horizontal
   }) : super(key: key) {
-    assert(key != null);
```

Comment:
I think removing this assert is OK but a little risky. It stems from: https://github.com/flutter/flutter/pull/2467#discussion_r55267809

Basically it was too hard to write an assert that detected when a missing key was needed and explained as much in a way that a developer could understand.  So we always required a key.  

If you fail to provide a key when it's needed, the failure mode is difficult to diagnose.